invenio/modules/accounts/models.py

```diff
@@ -135,8 +135,8 @@ def verify_password(self, password, migrate=False):
 
     def verify_email(self, force=False):
         """Verify email address."""
-        if force or self.note == "2":
-            if self.note != "2":
+        if force or str(self.note) == "2":
```

Comment:
can you rather fix the setter of `self.note` to always convert everything to `text_type`?